litestar/plugins/problem_details.py

```diff
@@ -75,14 +77,22 @@ def __init__(
         self.title = title
         self.instance = instance
 
-    def to_response(self, request: Request) -> Response:
+    def to_response(self, request: Request[Any, Any, Any]) -> Response[dict[str, Any]]:
         """Convert the problem details exception into a ``Response.``"""
 
         problem_details = {"type": self.type_, "status": self.status_code}
-        for attr in ("title", "instance", "detail", "extra"):
+        for attr in ("title", "instance", "detail"):
             if (value := getattr(self, attr, None)) is not None:
                 problem_details[attr] = value
 
+        if isinstance(self.extra, Mapping):
+            problem_details.update(self.extra)
+        elif (extra := self.extra) is not None:
+            if isinstance(extra, Mapping):
+                problem_details.update(extra)
+            else:
+                problem_details["extra"] = extra
+
```

Comment:
Isn't there some strange redundancy in these conditionals?
Both L88 and L91 check if self.extra is a Mapping, meaning that if it is a Mapping, L92 will never happen. Right?

To me, it looks like L91-93 can be removed and L94 can be unindented one step to replace the whole interior conditional.